src/lexer.coffee

```diff
@@ -666,8 +666,11 @@ exports.Lexer = class Lexer
         return 2
       else if firstChar is '{'
         if prevChar is ':'
-          token = @token '(', '('
+          token = @token '(', '{'
```

Comment:
Can you please explain what’s going on here? And if the `(` version originally wasn’t just a mistake, then maybe should we add an explanation in a comment?

Reply:
This case appears to correspond to a JSX attribute value that's an expression (eg the `{b}` in `<div a={b} />`)

This is just another example of how we only fake-parse JSX but now need to preserve the real source tokens as well - here, to our grammar this is a `(` since we're just wrapping the attribute value expression in parens, but in reality it's a `{` in the source

Reply:
```suggestion
          # This token represents the start of a JSX attribute value
          # that’s an expression (e.g. the `{b}` in `<div a={b} />`).
          # Our grammar represents the beginnings of expressions as `(`
          # tokens, so make this into a `(` token that displays as `{`.
          token = @token '(', '{'
```